modules/hardware/nvidia-jetson-orin/virtualization/host/bpmp-virt-host/default.nix

```diff
@@ -0,0 +1,44 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}: let
+  cfg = config.ghaf.hardware.nvidia.orin.virtualization.host;
+in {
+  options.ghaf.hardware.nvidia.orin.virtualization.host.enable = lib.mkOption {
```

Comment:
Looks good.

Proposing `ghaf.hardware.nvidia.bpmp.virtualization.host.enable` instead.

Reasoning: 
- This is NVIDIA BPMP (Boot and Power Management Processor) virtualization specific. Not general virtualization related. General virtualization works already.
- Also, proposing that we drop `orin` as the more generic for BPMP in this context would be `jetson` but that is unnecessary specific as well as these options are declared in the targets which make it very specific. 
(To be very exact, BPMP is specific to Tegra chip)

Reply:
> This is NVIDIA BPMP (Boot and Power Management Processor) virtualization specific. Not general virtualization related. General virtualization works already.

If we scope it this way, maybe `virtualization` is also redundant here. I wanted to create some scope where to put all configuration related to the passthroughs we are working on. Maybe we could just call it `passthroughs`? Something like:

```
ghaf.hardware.nvidia.passthroughs = {
  enable = true; # Enables bpmp virtualization and other necessary scaffolding.
  
  # If we are defining a host configuration (mutually exclusive with guest.enable = true)
  host = {
    enable = true;
    uarta = true;
    gpu = true;
    ...
  };
  
  # If we are defining a guest configuration (mutually exclusive with host.enable = true)
  guest = {
    enable = true;
    uarta = true;
    gpu = true;
  };
};
```

> Also, proposing that we drop orin as the more generic for BPMP in this context would be jetson but that is unnecessary specific as well as these options are declared in the targets which make it very specific.
(To be very exact, BPMP is specific to Tegra chip)

Ok yeah this make sense if the assumption is that these changes should work with all the Tegra chips.

Reply:
> > This is NVIDIA BPMP (Boot and Power Management Processor) virtualization specific. Not general virtualization related. General virtualization works already.
> 
> If we scope it this way, maybe `virtualization` is also redundant here. I wanted to create some scope where to put all configuration related to the passthroughs we are working on. 

I think `virtualization` here is warranted. These drivers (host+guest) are more virtualization and this is not passthrough. Technically passthrough could be considered not virtualization (though academic-engineering definitions vary) but we often have "hybrid-virtualization" where CPU(s) and memory are virtualized and some device is just passed through as native - with native driver in the guest. Here with BPMP we actually have both backend driver and frontend driver but it's not virtualized in the sense of making the hardware specifics agnostic - like virtio. Making this more generic with other similar future implementations in mind even may not even make sense on something this hardware/vendor specific. IMO this is more virtualization than classical passthrough.

> Maybe we could just call it `passthroughs`? Something like:

See above comment ...
> ```
> ghaf.hardware.nvidia.passthroughs = {
>   enable = true; # Enables bpmp virtualization and other necessary scaffolding.
>   
>   # If we are defining a host configuration (mutually exclusive with guest.enable = true)
>   host = {
>     enable = true;
>     uarta = true;
>     gpu = true;
>     ...
>   };
>   
>   # If we are defining a guest configuration (mutually exclusive with host.enable = true)
>   guest = {
>     enable = true;
>     uarta = true;
>     gpu = true;
>   };
> };
> ```
I like your proposed structure on host and guest though I'm not sure in the internal implementation if it's even possible to only enable some but not all under host or guest. Better check that with Vadim and J-P.
But it would be good to have this top-level enabling option:
```
 ghaf.hardware.nvidia.passthroughs = {
   enable = true;
``` 
with implementation broken down at least to `host` and `guest.`. I'm afraid Nix won't allow us to leave those options as internal or private but we can guide ghaf external users to use only the top-level option and mark the internal options `host` and `guest` as implementation details with the purpose of incremental integration, debugging and testing.

> > Also, proposing that we drop orin as the more generic for BPMP in this context would be jetson but that is unnecessary specific as well as these options are declared in the targets which make it very specific.
> > (To be very exact, BPMP is specific to Tegra chip)
> 
> Ok yeah this make sense if the assumption is that these changes should work with all the Tegra chips.

Reply:
Refactored and documented options in [this commit](https://github.com/tiiuae/ghaf/pull/394/commits/00f4754727088500ebb1f9c254f75876ba3652fe),

New options:

```
# Enables common virtualization modules (implicit option enabled by other options).
ghaf.hardware.nvidia.virtualization.enable

# Enables host virtualization modules implicit option enabled by other options).
ghaf.hardware.nvidia.virtualization.host.bpmp.enable

# Enables UARTA passthrough on host, enables the above implicit options.
ghaf.hardware.nvidia.passthroughs.host.uarta.enable
```

Documentation updated to reflect implicit options.